src/basic/process-util.h

```diff
@@ -179,3 +179,18 @@ static inline int safe_fork(const char *name, ForkFlags flags, pid_t *ret_pid) {
 }
 
 int fork_agent(const char *name, const int except[], unsigned n_except, pid_t *pid, const char *path, ...);
+
+#if SIZEOF_PID_T == 4
+#define PID_T_MAX ((pid_t) INT32_MAX)
+/* The maximum number of PIDs Linux allows on this architecture. Tthis reflects a limit compiled into the kernel, and
+ * sets the limit on what may be written to the /proc/sys/kernel/pid_max sysctl (but do note that the sysctl is off by
+ * 1, since PID 0 can never exist). */
+#define TASKS_MAX 4194303U
+#elif SIZEOF_PID_T == 2
+#define PID_T_MAX ((pid_t) INT16_MAX)
+#define TASKS_MAX 32767U
+#else
+#error "Unknown pid_t size"
+#endif
+
+assert_cc((unsigned long) TASKS_MAX <= (unsigned long) PID_T_MAX)
```

Comment:
Why the cast? Shouldn't we just let the compiler up-cast to the longer unsigned integer?

Reply:
well, i have to cast the PID_T_MAX value to something because otherwise we'll compare unsigned and signed and that is not an OK thing to do with our compiler settings... sure, i wouldn't have to cast the TASKS_MAX value (because it already is unsigned), but it made more sense to me to cast that too...